### PR TITLE
this will add sort functionality - comment count not available on api

### PR DESCRIPTION
**Pull request by WolfieKnee**
All available sort options from the api are included, but comment count wasn't included as part of that project so putting on backlog...

### CODE REVIEW

File: src/components/SortArticles.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		<form>
+			<fieldset>
+				<label htmlFor="sort_by">sort by: </label>
+				<select name="sort_by" value={sortBy} onChange={handleSortBy}>
+					<option value="created_at">date</option>
+					<option value="title">title</option>
+					<option value="topic">topic</option>
+					<option value="author">author</option>
+					<option value="votes">votes</option>
+					{/* <option value="commentCount">comment count</option> */}
```
Comment (nc-richard-neat) [inline src/components/SortArticles.jsx:32]: pls remove all comments and logs before committing

File: src/pages/ArticleList.jsx [hunk 2]
```diff
@@ -2,22 +2,32 @@ import { useEffect, useState } from "react";
...
+		if (sortBy) {
+			apiQuery.append("sort_by", sortBy);
+		}
+		if (sortOrder) {
+			apiQuery.append("order", sortOrder);
+		}
+
 		setIsLoading(true);
-		newsAPIGet(url)
+		newsAPIGet(`${url}?${apiQuery}`)
```
Comment (nc-richard-neat) [inline src/pages/ArticleList.jsx:30]: You could make sure of axios' ability to handle optional params here, if you pass through the values for the sorting and topics etc and if they're undefined axios will simply ignore them. Take a look at the axios notes for help 👍

**Review by nc-richard-neat — COMMENTED** [review]
Looking really nice, great choice here to update the url with the queries too.
Just a couple of things I noticed:
- When I click 'home' the url goes back to default but the page doesn't update with no sorting options.
- The default sort order doesn't appear in the url unless I click ascending then back to descending.

Minor really but worth taking a look at if you have time. Nice work :)

File: src/pages/ArticleList.jsx (hunk 2 above)
Comment (WolfieKnee) [reply src/pages/ArticleList.jsx:30]: Great tip - that's much cleaner :smile:

**Pull request merged**